sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -311,6 +311,7 @@ Azure::Core::Http::CurlTransportOptions CurlTransportOptionsFromTransportOptions
     curlOptions.SslOptions.PemEncodedExpectedRootCertificates
         = PemEncodeFromBase64(transportOptions.ExpectedTlsRootCertificate, "CERTIFICATE");
   }
+  curlOptions.SslVerifyPeer = !transportOptions.DisableTlsCertificateValidation;
```

Comment:
Just making sure (because it seems to be ok), but @LarryOsterman would we need to update the connection key logic that we use for the connection pooling/caching? It is meant to be unique for each set of options (based on `SslVerifyPeer` being set).
https://github.com/Azure/azure-sdk-for-cpp/blob/e2a1b99a0625d6309e22474332fb3c342b4a2dc6/sdk/core/azure-core/src/http/curl/curl.cpp#L1293

Reply:
Good question: Based on my reading, the SslVerifyPeer option is already included in the connection key logic - see line 1284 of `GetConnectionKey`.